src/app/app.component.ts

```diff
@@ -0,0 +1,57 @@
+import { Component, OnInit } from '@angular/core';
+import { CoffeeShopsService } from './services/coffeeShops.service';
+import {  retry, switchMap } from 'rxjs';
+import { CoffeeShopModel } from './models/coffeeShop.model';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.scss']
+})
+export class AppComponent implements OnInit {
+  coffeeShops = new Array<CoffeeShopModel>();
+  coordinatesFormControl = new FormGroup({
+    xCoordinate: new FormControl([], Validators.required),
+    yCoordinate: new FormControl([], Validators.required)
+  });
+
+  constructor(private coffeeShop: CoffeeShopsService) {
+  }
+
+  ngOnInit(): void {
+    this.getCoffeeShops();
+  }
+
+  getCoffeeShops(): void {
+    this.coffeeShop.getToken().pipe(
+      switchMap((result: any) => {
+        return this.coffeeShop.getCoffeeShops(result.token);
+      }), retry(5)).subscribe(result => this.coffeeShops = result);
+
+  }
+
+  getNearbyCS(): void {
+    for (let coffeeShop of this.coffeeShops) {
+      coffeeShop.distanceFromUser = Number(this.distance(this.coordinatesFormControl.value.xCoordinate, this.coordinatesFormControl.value.yCoordinate,
+        Number(coffeeShop.x), Number(coffeeShop.y)));
+    }
```

Comment:
This function call is fundamentally very simple, it calculates the distance based on 2 Xs and 2 Ys. But because you are accessing the values form the `formControls` directly, it ends up spanning 2 lines. Can you rewrite this somehow to be more clear what is going on? I think destructuring the fields values would help a ton.

Reply:
> Overall the App looks nice and works fine despite the UI not being required. I have two final questions though:
> 
> * Regarding the tests, was it just a time constraint or is testing not part of your usual development cycle?
> * Why put yourself in the situation to write the UI, and add all these extra dependencies when the provided starting app had almost all the parts required to do the CLI way? The time spent making the UI could have been used to write very simple unit tests.

* Regarding the tests, it's a bit of both. It was a time constraint. I know how to write tests, but in my daily job I am not required to write them, so I am a bit out of shape regarding testing. I am working on implementing some tests right now, and will commit in a some time. 
* I wrote the UI because I am more of a visual person, and even if the bare minimum was to create a console / cli, it was more natural for me to put together a basic ui. I tried to do it in react or node, but I got frustrated and decided to go with what I know, Angular. That's why I encountered the time contraint.